### PR TITLE
[ICDS Dashboard] asana task: updated disclaimer messages in icds dashboard

### PR DESCRIPTION
**Pull request by dhiraj-beehyv**
[Asana Task](https://app.asana.com/0/1134847685842845/1134020518211440)

Updated the ICDS dashboard disclaimer messages as per the spec.

### CODE REVIEW

**Comment by dhiraj-beehyv** [comment]
@Rohit25negi ,

Can you review this pr as well. Thanks

File: custom/icds_reports/static/js/directives/info-message/info-message.directive.js [hunk 1]
```diff
@@ -4,21 +4,33 @@ var url = hqImport('hqwebapp/js/initial_page_data').reverse;
...
         var selectedYear = parseInt($location.search()['year']) || new Date().getFullYear();
         var currentMonth = new Date().getMonth() + 1;
         var currentYear = new Date().getFullYear();
-        if (!$location.path().startsWith("/fact_sheets") && !$location.path().startsWith("/download") &&
-            selectedMonth === currentMonth && selectedYear === currentYear &&
-            (new Date().getDate() === 1 || new Date().getDate() === 2)) {
-            vm.lastDayOfPreviousMonth = moment().set('date', 1).subtract(1, 'days').format('Do MMMM, YYYY');
+        var currentDate = new Date().getDate();
+
+        if (selectedMonth === currentMonth && selectedYear === currentYear && (currentDate >= start && currentDate <= end)) {
```
Comment (czue) [inline custom/icds_reports/static/js/directives/info-message/info-message.directive.js]: this complex conditional could be well-suited to extraction (ether as a variable or function). e.g.

```
// not sure if this is the right name
var dateIsInCurrentMonth = (selectedMonth === currentMonth && selectedYear === currentYear && (currentDate >= start && currentDate <= end))
if (dateIsInCurrentMonth) {
...
}
```

File: custom/icds_reports/templates/icds_reports/icds_app/cas-export.directive.html [hunk 2]
```diff
@@ -4,7 +4,8 @@
             <span class="breadcrump-element">Export CAS Data: </span>
         </div>
     </div>
-    <info-message></info-message>
+    <info-message start="1" end="6" type="download-7"></info-message>
+    <info-message start="7" end="31" type="download-7-last"></info-message>
```
Comment (czue) [inline custom/icds_reports/templates/icds_reports/icds_app/cas-export.directive.html]: based on the name should this not be `start="24"` or something? (caveat: I have no idea how these work)

Comment (czue) [reply custom/icds_reports/templates/icds_reports/icds_app/cas-export.directive.html]: looked at the spec and think the logic is right. but maybe the "type" is oddly named? what were you thinking `download-7-last` meant?

File: custom/icds_reports/templates/icds_reports/icds_app/info-message.directive.html [hunk 3]
```diff
@@ -1,5 +1,22 @@
...
+      </span>
+      <span ng-if="$ctrl.type === 'download-7-last'">
+        You are downloading the data for {$ $ctrl.previousMonth $}, {$ $ctrl.year $}. The data for {$ $ctrl.currentMonth $}, {$ $ctrl.year $} will be available beginning {$ $ctrl.nextMonth $} 7th, {$ $ctrl.year $}.
+      </span>
+      <span ng-if="$ctrl.type === 'download-15-last'">
+        You are downloading the data for {$ $ctrl.previousMonth $}, {$ $ctrl.year $}. The data for {$ $ctrl.currentMonth $}, {$ $ctrl.year $} will be available beginning {$ $ctrl.nextMonth $} 15th, {$ $ctrl.year $}.
+      </span>
+      <span ng-if="$ctrl.type === 'default'">
+        You are viewing the data for {$ $ctrl.previousMonth $}, {$ $ctrl.year $}. The data for {$ $ctrl.currentMonth $}, {$ $ctrl.year $} will be available beginning  {$ $ctrl.currentMonth $} 3rd, {$ $ctrl.year $}.
+      </span>
```
Comment (czue) [inline custom/icds_reports/templates/icds_reports/icds_app/info-message.directive.html]: I didn't check these carefully but assume you got them right from the spec

**Review by czue — COMMENTED** [review]
fyi i'm not approving or requesting changes since I'm still coming on-board. will leave that to @Rohit25negi and others for now.

File: custom/icds_reports/static/js/directives/info-message/info-message.directive.js (hunk 1 above)
Comment (dhiraj-beehyv) [reply custom/icds_reports/static/js/directives/info-message/info-message.directive.js]: @czue ,
You're right, I'll separate the condition to a var.

File: custom/icds_reports/templates/icds_reports/icds_app/cas-export.directive.html (hunk 2 above)
Comment (dhiraj-beehyv) [reply custom/icds_reports/templates/icds_reports/icds_app/cas-export.directive.html]: I've updated the type and made them more verbose, do let me know if you have any thoughts on it.

Comment (czue) [reply custom/icds_reports/templates/icds_reports/icds_app/cas-export.directive.html]: looks good, thanks

**Review by czue — APPROVED** [review]
g2g from my side

File: custom/icds_reports/templates/icds_reports/icds_app/cas-export.directive.html [hunk 4]
```diff
@@ -4,7 +4,8 @@
             <span class="breadcrump-element">Export CAS Data: </span>
         </div>
     </div>
-    <info-message></info-message>
```
Comment (dhiraj-beehyv) [inline custom/icds_reports/templates/icds_reports/icds_app/cas-export.directive.html]: This was not needed, under the directive it was hidden for the download page by checking the active url explicitly. Therefore I remove this from here and cleaned up code in info-message.directive.js.

**Review by Rohit25negi — APPROVED** [review]

**Pull request merged**